autorest/azure/rp.go

```diff
@@ -0,0 +1,188 @@
+package azure
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// DoRetryWithRegistration tries to register the resource provider in case it is unregistered.
+// It also handles request retries
+func DoRetryWithRegistration(client autorest.Client) autorest.SendDecorator {
+	return func(s autorest.Sender) autorest.Sender {
+		return autorest.SenderFunc(func(r *http.Request) (resp *http.Response, err error) {
+			attempts := 0
+			rr := autorest.NewRetriableRequest(r)
+			for attempts < client.RetryAttempts {
```

Comment:
[nit] I'd rename `attempts` to something that didn't sound static like `currentAttempt`. Also, it would be a little more readable/conventional to declare `attempts` in the for body like so:

``` Go
for currentAttempt := 0; currentAttempt < client.RetryAttempts; currentAttempt++ {
```

Reply:
Fixed